sklearn/linear_model/tests/test_base.py

```diff
@@ -105,12 +105,14 @@ def test_raises_value_error_if_positive_and_sparse():
         reg.fit(X, y)
 
 
-def test_raises_value_error_if_sample_weights_greater_than_1d():
+def test_raises_value_error_if_sample_weights_greater_than_1d(random_state=0):
```

Comment:
removed the `global_random_seed` because we only test if the `sample_weight`s that should work, actually do - or raise an (unintended) `ValueError`. Function that checks the weights: `scikit-learn/sklearn/utils/validation.py:_check_sample_weight` is tested in `sklearn/utils/tests/test_validation.py:test_check_sample_weight`

Reply:
We can remove `random_state=0` from the signature and instead define it within the test body

Reply:
Moved to global `rng`

Reply:
defined `rng = np.random.RandomState(0)` within test body for all tests that do not need `global_random_seed`